src/ucp/wireup/select.c

```diff
@@ -1203,10 +1204,16 @@ ucp_wireup_add_amo_lanes(const ucp_wireup_select_params_t *select_params,
         }
     }
 
-    return ucp_wireup_add_memaccess_lanes(select_params, ep_init_flags,
-                                          &criteria, UCS_MEMORY_TYPE_HOST,
-                                          tl_bitmap, UCP_LANE_TYPE_AMO,
-                                          select_ctx);
+    ucs_memory_type_for_each(mem_type) {
+        ucs_status_t status = ucp_wireup_add_memaccess_lanes(
+                select_params, ep_init_flags, &criteria, mem_type, tl_bitmap,
+                UCP_LANE_TYPE_AMO, select_ctx);
+        if ((status != UCS_OK) && (mem_type == UCS_MEMORY_TYPE_HOST)) {
+            return status;
```

Comment:
I think it is better to add a debug message to the log when `(status != UCS_OK) && (mem_type != UCS_MEMORY_TYPE_HOST)`.